PolyEngine/Engine/Src/AssetsPathConfig.cpp

```diff
@@ -1,140 +1,35 @@
 #include "EnginePCH.hpp"
 
-#include "rapidjson/document.h"
+RTTI_DEFINE_TYPE(Poly::AssetsPathConfig)
 
 namespace Poly
 {
-	AssetsPathConfig gAssetsPathConfig;
+	static const String DEFAULT_ENGINE_ASSETS_PATH = String("../../Engine/Res/");
+	static const String DEFAULT_GAME_ASSETS_PATH = String("../../Games/SGJGame/Res/");
 
-	Dynarray<String> AssetsPathConfig::GetAssetsPaths(eResourceSource Source) const
-	{
-		HEAVY_ASSERTE(Source < eResourceSource::_COUNT, "_COUNT enum value passed to GetAssetsPaths(), which is an invalid value");
-		// TODO: move to very first start block in program
-		if (!gAssetsPathConfig.IsLoadedFromFile())
-			gAssetsPathConfig.ReloadFromFile();
-		
-		return AssetsPaths[Source];
-	}
+	AssetsPathConfig gAssetsPathConfig;
 
-	bool AssetsPathConfig::IsLoadedFromFile() const
+	AssetsPathConfig::AssetsPathConfig() : ConfigBase("Assets Path Config", eResourceSource::NONE)
 	{
-		return LoadedFromFile;
+		EngineAssetsPath = DEFAULT_ENGINE_ASSETS_PATH;
+		GameAssetsPath = DEFAULT_GAME_ASSETS_PATH;
 	}
 
-	bool AssetsPathConfig::ReloadFromFile()
+	const String& AssetsPathConfig::GetAssetsPath(eResourceSource source) const
 	{
-		gConsole.LogInfo("CoreConfig::ReloadFromFile() called");
-		bool IsSuccessfull = true;
-
-		String ConfigFileContent;
-		try
-		{
-			ConfigFileContent = LoadTextFile(ASSETS_PATH_CONFIG_PATH);
-			// gConsole.LogInfo("File Content: {}", ConfigFileContent);
-		}
-		catch (FileIOException)
-		{
-			IsSuccessfull = false;
-			gConsole.LogInfo("Config file reading failed: {}, using defaults", ASSETS_PATH_CONFIG_PATH);
-
-			LoadDefaults();
-
-			try
-			{
-				WriteDefaultJson();
-			}
-			catch (FileIOException)
-			{
-				gConsole.LogInfo("Config file saving defaults failed");
-			}
-		}
-
-		rapidjson::Document Json; // UTF8 by default
-		if (Json.Parse(ConfigFileContent.GetCStr()).HasParseError())
-		{
-			IsSuccessfull = false;
-			gConsole.LogInfo("Config file json has error: {}", Json.GetParseError());
-
-			LoadDefaults();
-		}
-		else
+		static String EMPTY;
```

Comment:
Why is this static and not a global constant?

Reply:
good idea!